pkg/operator/cluster/ceph/osd/osd.go

```diff
@@ -189,13 +204,26 @@ func (c *Cluster) Start() error {
 				}
 			}
 		} else {
-			logger.Infof("osd replica set started for node %s", n.Name)
+			logger.Infof("osd pod started for node %s", n.Name)
 		}
 
 		// wait for the current node's orchestration to be completed
-		if err := c.waitForCompletion(n.Name); err != nil {
+		if status, err := c.waitForCompletion(n.Name); err != nil {
+			logger.Infof("failed to start osd: %v", err)
 			errorMessages = append(errorMessages, err.Error())
 			continue
+		} else {
+			// start OSD daemon
+			if status != nil {
+				for _, osd := range status.OSDs {
+					logger.Infof("creating OSD :%+v", osd)
+					rs := c.makeOSDReplicaSet(n.Name, c.resources, osd)
```

Comment:
can we use deployments instead of replicasets? As seen in #1431, deployments are preferable so this would be a good time to switch the OSDs.